### PR TITLE
rough draft of README

### PR DESCRIPTION
**Pull request by burrrata**
Created a rough draft of the main README in the dissent-voting-app repo as well as the READMEs of the Dissent Oracle and Dandelion Voting apps. 

Mostly just want to make sure that I understand what's going on. Please weigh in with suggestions for improvement or just go ahead and make changes as necessary.

### CODE REVIEW

File: dandelion-voting/README.md [hunk 1]
```diff
@@ -22,9 +56,22 @@ npm install
...
+
+## Installing to an Aragon DAO
+
+TBD
+
+## Contributing
+
+We welcome community contributions!
+
+Please check out our [open Issues](https://github.com/1Hive/time-lock-app/issues) to get started.
```
Comment (fabriziovigevani) [inline dandelion-voting/README.md]:
```suggestion
Please check out our [open Issues](https://github.com/1Hive/dissent-voting-app/issues) to get started.
```

File: dissent-oracle/README.md [hunk 2]
```diff
@@ -1,9 +1,47 @@
...
+# Dissent Oracle <img align="right" src="https://github.com/1Hive/website/blob/master/website/static/img/bee.png" height="80px" />
 
-Oracle implementation installed along with the Dandelion Voting app to prevent engagement when a user has recently voted in favour of a proposal.
+The Dissent Oracle app is an [ACL Oracle](https://hack.aragon.org/docs/acl_IACLOracle). ACL Oracles are small helper functions that plug in to Aragon's access control list (ACL) to do more sophisticated permission evaluation. The Dissent Oracle is installed along with the Dandelion Voting app to prevent engagement when a user has recently voted in favour of a proposal.
 
-## Usage
+#### 🐲 Project stage: development
 
-Applied to an Aragon app's permission's params.
+The Time Lock app is still in development, a first implementation was published to `time-lock.open.aragonpm.eth`. If you are interested in contributing please see our open [issues](https://github.com/1hive/time-lock-app/issues).
```
Comment (fabriziovigevani) [inline dissent-oracle/README.md]: Should be removed

File: dissent-oracle/README.md [hunk 3]
```diff
@@ -1,9 +1,47 @@
...
+
+#### 🚨 Security review status: pre-audit
+
+The code in this repo has not been audited.
+
+## How does it work?
+
+The Dissent Oracle is initialized with a dissent window. The dissent window is the window of time in which an account cannot dissent (exit the Dandelion Org) if they have performed an action (such as voting `yes` for a proposal). The Dandelion Voting app queries the Dissent Oracle to check if an account voted `yes` within the dissent window. In the context of Dandelion Orgs this is used to check if an account can or cannot redeem tokens from the Redemptions app.
```
Comment (fabriziovigevani) [inline dissent-oracle/README.md]:
```suggestion
The Dissent Oracle is initialized with a dissent window. The dissent window is the window of time in which an account cannot dissent (exit the Dandelion Org) if they have performed an action (such as voting `yes` for a proposal). The Dandelion Voting app queries the Dissent Oracle (through the ACL) to check if an account voted `yes` within the dissent window. In the context of Dandelion Orgs this is used to check if an account can or cannot redeem tokens from the Redemptions app.
```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,33 @@
-# dissent-app
-The dissent app enables organizations to restrict actions to only those who have not expressed approval in recent votes.
+# Dissent-Voting <img align="right" src="https://github.com/1Hive/website/blob/master/website/static/img/bee.png" height="80px" />
+
+The dissent-voting app enables organizations to restrict voting to those who have not expressed approval in recent votes. This is achieved through a combination of the [Dandelion Voting](https://github.com/1Hive/dissent-voting-app/tree/master/dandelion-voting) and [Dissent Oracle](https://github.com/1Hive/dissent-voting-app/tree/master/dissent-oracle). Dandelion Voting keeps track of who has voted `yay` or `nay` on each proposal and when that vote took place. The Dissent Oracle then checks whether an account has voted `yay` on a proposal within the time-frame of the dissent-window and returns a boolean expressing that fact. This allows other applications (such as [Redemptions](https://github.com/1Hive/redemptions-app/)) to restrict actions based on the response from the Dissent Oracle.
```
Comment (lkngtn) [inline README.md]: > or `nay`

We only keep track of `yay` votes afaik.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,33 @@
...
+
+The Dissent-Voting application is still in development. If you are interested in contributing please see our open [issues](https://github.com/1hive/dissent-voting-app/issues).
+
+#### 🚨 Security review status: pre-audit
+
+The code in this repo has not been audited.
+
+## How does it work?
+
+Details on the mechanics of the [Dandelion Voting]() and [Dissent Oracle]() apps can be found in the README of each app:
```
Comment (lkngtn) [inline README.md]: Links appear to be broken

File: dissent-oracle/README.md (hunk 2 above)
Comment (lkngtn) [inline dissent-oracle/README.md]: This is for the dissent oracle not for time-lock no?

File: dissent-oracle/README.md [hunk 6]
```diff
@@ -1,9 +1,47 @@
...
+
+## Installing to an Aragon DAO
+
+TBD
+
+## Contributing
+
+We welcome community contributions!
+
+Please check out our [open Issues](https://github.com/1Hive/time-lock-app/issues) to get started.
```
Comment (fabriziovigevani) [inline dissent-oracle/README.md]:
```suggestion
Please check out our [open Issues](https://github.com/1Hive/dissent-voting-app/issues) to get started.
```

File: dissent-oracle/README.md [hunk 7]
```diff
@@ -1,9 +1,47 @@
...
+
+## How does it work?
+
+The Dissent Oracle is initialized with a dissent window. The dissent window is the window of time in which an account cannot dissent (exit the Dandelion Org) if they have performed an action (such as voting `yes` for a proposal). The Dandelion Voting app queries the Dissent Oracle to check if an account voted `yes` within the dissent window. In the context of Dandelion Orgs this is used to check if an account can or cannot redeem tokens from the Redemptions app.
+
+> Note: this will not work with the regular Aragon voting application because it does not track the last block in which a voter voted yea.
+
+## Initialization
+
+The Dissent Oracle app is initialized with an `address _dandelionVoting` and `uint64 _dissentWindowBlocks`. The `address _dandelionVoting` is the address of the voting app that the Dissent Oracle is to query. The `uint64 _dissentWindowBlocks` is the time window within which the dissent oracle should return a boolean.
```
Comment (lkngtn) [inline dissent-oracle/README.md]: > is the time window within

Probably should say is the time window (expressed as a number of blocks) within which the dissent oracle should return a boolean.

Comment (fabriziovigevani) [inline dissent-oracle/README.md]: I would change the `should return a boolean` for something less technical maybe?

File: dissent-oracle/README.md (hunk 2 above)
Comment (burrrata) [reply dissent-oracle/README.md]: Yup. Lots of copypasta for the READMEs and lots of link errors. Thanks for catching that. 

Also, FTR created a README template in devops to address that in the future: https://github.com/1Hive/devops/blob/master/docs/README.md

File: README.md (hunk 5 above)
Comment (burrrata) [reply README.md]: Good catch. The links are actually right below this line so just removed these extra ones.

File: README.md (hunk 4 above)
Comment (burrrata) [reply README.md]: Good call! Thanks

File: dissent-oracle/README.md (hunk 2 above)
Comment (burrrata) [reply dissent-oracle/README.md]: Fixed

File: dissent-oracle/README.md (hunk 7 above)
Comment (burrrata) [reply dissent-oracle/README.md]: Updated to explain that the dissent oracle returns true if an account voted yes within the dissent window. Is that better?

**Comment by fabriziovigevani** [comment]
Closing in favour of #42

**Pull request closed without merging**